File: viajes365-backend/Viajes365RestApi/Entities/Tour.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;


namespace Viajes365RestApi.Entities
{
    public class Tour : Base
    {
        [Key]
        public long TourId { get; set; }
        [StringLength(100)]
        [Required]
        public string Name { get; set; }
        public string Summary { get; set; }
        [StringLength(20)]
        public string Duration { get; set; }

        [ForeignKey("Location")]
        public long LocationId { get; set; }
        public virtual Location Location { get; set; }

        public virtual ICollection<Attraction> Attractions { get; set; }

        public virtual ICollection<Photo> Photos { get; set; }

    }
}

File: viajes365-frontend/src/app/_models/tour.ts
import { Attraction } from './attraction';
import { Photo } from './photo';

export class Tour {
  tourId!: number;
  name!: string;
  summary!: string;
  duration!: string;
  locationId!: number;
  attractions!: Attraction[];
  photos!: Photo[];
  active!: boolean;

}
